llvm/lib/Target/AMDGPU/AMDGPUGlobalISelUtils.cpp

```diff
@@ -56,7 +56,7 @@ AMDGPU::getBaseWithConstantOffset(MachineRegisterInfo &MRI, Register Reg,
 
   Register Base;
   if (KnownBits && mi_match(Reg, MRI, m_GOr(m_Reg(Base), m_ICst(Offset))) &&
-      KnownBits->maskedValueIsZero(Base, APInt(32, Offset)))
+      KnownBits->maskedValueIsZero(Base, APInt(32, Offset, /*isSigned=*/true)))
```

Comment:
APSint?

Reply:
Jay suggested to use APInt in the same case use https://github.com/llvm/llvm-project/pull/122251#discussion_r1908574869

@jayfoad?

Reply:
What is the suggestion? I don't even see how you could use APSInt here.

Reply:
Oh, indeed, that was just an incorrect usage of `APSInt(BitWidth, isUnsigned)`...